applications/StructuralMechanicsApplication/custom_constitutive/linear_j2_plasticity_3d.cpp

```diff
@@ -174,95 +180,123 @@ void LinearJ2Plasticity3D::CalculateMaterialResponseKirchhoff(ConstitutiveLaw::P
-                                       stress_trial_dev(2) * stress_trial_dev(2) +
-                                       2. * stress_trial_dev(3) * stress_trial_dev(3) +
-                                       2. * stress_trial_dev(4) * stress_trial_dev(4) +
-                                       2. * stress_trial_dev(5) * stress_trial_dev(5));
-    trial_yield_function = this->YieldFunction(norm_dev_stress, rMaterialProperties);
-
-    if (trial_yield_function <= 0.) {
-        // ELASTIC
-        mInelasticFlag = false;
-        stress_vector = yield_tensionrial;
-        tangent_tensor = elastic_tensor;
-    } else {
-        // INELASTIC
-        mInelasticFlag = true;
-        double dgamma = 0;
-        Vector yield_function_normal_vector = stress_trial_dev / norm_dev_stress;
-        if (delta_k != 0.0 && hardening_exponent != 0.0) {
-            // Exponential softening
-            dgamma = GetDeltaGamma(norm_dev_stress, rMaterialProperties);
-        }
-        else {
-            // Linear softening
-            dgamma = trial_yield_function /
-                     (2. * mu * (1. + (hardening_modulus / (3. * mu))));
-        }
 
-        stress_vector(0) =
-            volumetric_modulus * (strain_vector(0) + strain_vector(1) + strain_vector(2)) +
-            stress_trial_dev(0) - 2. * mu * dgamma * yield_function_normal_vector(0);
-        stress_vector(1) =
-            volumetric_modulus * (strain_vector(0) + strain_vector(1) + strain_vector(2)) +
-            stress_trial_dev(1) - 2. * mu * dgamma * yield_function_normal_vector(1);
-        stress_vector(2) =
-            volumetric_modulus * (strain_vector(0) + strain_vector(1) + strain_vector(2)) +
-            stress_trial_dev(2) - 2. * mu * dgamma * yield_function_normal_vector(2);
-        stress_vector(3) =
-            stress_trial_dev(3) - 2. * mu * dgamma * yield_function_normal_vector(3);
-        stress_vector(4) =
-            stress_trial_dev(4) - 2. * mu * dgamma * yield_function_normal_vector(4);
-        stress_vector(5) =
-            stress_trial_dev(5) - 2. * mu * dgamma * yield_function_normal_vector(5);
-
-        mPlasticStrain(0) = mPlasticStrainOld(0) + dgamma * yield_function_normal_vector(0);
-        mPlasticStrain(1) = mPlasticStrainOld(1) + dgamma * yield_function_normal_vector(1);
-        mPlasticStrain(2) = mPlasticStrainOld(2) + dgamma * yield_function_normal_vector(2);
-        mPlasticStrain(3) = mPlasticStrainOld(3) + dgamma * yield_function_normal_vector(3) * 2;
-        mPlasticStrain(4) = mPlasticStrainOld(4) + dgamma * yield_function_normal_vector(4) * 2;
-        mPlasticStrain(5) = mPlasticStrainOld(5) + dgamma * yield_function_normal_vector(5) * 2;
-        mAccumulatedPlasticStrain = mAccumulatedPlasticStrainOld + sqrt_two_thirds * dgamma;
+    // If we compute the tangent moduli or the stress
+    if( r_constitutive_law_options.Is( ConstitutiveLaw::COMPUTE_STRESS ) ||
```

Comment:
This "if" is fine, but requires that SetStress vector and SetConstitutiveMatrix are set in the calls from the elements, regardless of the values of COMPUTE_STRESS and COMPUTE_CONSTITUTIVE_TENSOR options. 

Currently it is missing in the CalculateAll function of the Bbar element (line ~132) and it will fail.

Reply:
Sorry I just saw the comments, I merged first, can we fix this in a hotfix?

Reply:
Sure. Well be working on these files anyways.